behaviour/graql/language/match.feature

```diff
@@ -191,68 +183,897 @@ Feature: Graql Match Clause
+      """
+    And concept identifiers are
+      |     | check | value |
+      | SUS | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | SUS |
+
+
+  Scenario: `has $attr !== $x` matches owners of any instance `$y` of `$attr` where `$y !== $x`
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Susie", has age 16, has ref 0;
+      $y isa person, has name "Donald", has age 25, has ref 1;
+      $z isa person, has name "Ralph", has age 18, has ref 2;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has age !== 18; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | DON | key   | ref:1 |
+      | SUS | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | DON |
+      | SUS |
+
+
+  Scenario: `has $attr > $x` matches owners of any instance `$y` of `$attr` where `$y > $x` even if they also own instance `$z` where `$z < $x`
+    Given graql define
+      """
+      define
+      lucky-number sub attribute, value long;
+      person has lucky-number;
+      """
+    Given the integrity is validated
+    Given graql insert
+      """
+      insert
+      $x isa person, has lucky-number 10, has lucky-number 20, has lucky-number 30, has ref 0;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match $x has lucky-number > 25; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | PER | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | PER |
+
+
+  ############
+  # PATTERNS #
+  ############
+
+  Scenario: disjunctions return the union of composing query statements
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Jeff", has ref 0;
+      $y isa company, has name "Amazon", has ref 1;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match {$x isa person;} or {$x isa company;}; get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | JEF | key   | ref:0 |
+      | AMA | key   | ref:1 |
+    Then uniquely identify answer concepts
+      | x   |
+      | JEF |
+      | AMA |
+
+
+  Scenario: a negation matches if the negated block has no matches
+    Given graql insert
+      """
+      insert
+      $x isa person, has name "Jeff", has ref 0;
+      """
+    Given the integrity is validated
+    When get answers of graql query
+      """
+      match
+        $x isa person;
+        not {
+          $e (employee: $x) isa employment;
+        };
+      get;
+      """
+    And concept identifiers are
+      |     | check | value |
+      | JEF | key   | ref:0 |
+    Then uniquely identify answer concepts
+      | x   |
+      | JEF |
+
+
+  Scenario: a negation does not match if the negated block has any matches
```

Comment:
we can talk more about negation i think with @jmsfltchr  and what we want to do to expand these cases!

Reply:
Don't fret about it now though, these are fine :)